File: pkg/AMORE/src/SimpleContainerIterator.cpp
#include "dia/SimpleContainerIterator.h"

template<typename T>
  SimpleContainerIterator<T>::SimpleContainerIterator()
  {
  }

template<typename T>
  SimpleContainerIterator<T>::~SimpleContainerIterator()
  {
  }

template<typename T>
  void
  SimpleContainerIterator<T>::first()
  {
    d_current = 0;
  }

template<typename T>
  void
  SimpleContainerIterator<T>::next()
  {
    ++d_current;
  }

template<typename T>
  bool
  SimpleContainerIterator<T>::isDone()
  {
    bool IteratorIsDone(d_current == d_container->size());
    return IteratorIsDone;
  }

template<typename T>
  T
  SimpleContainerIterator<T>::currentItem()
  {
      if (isDone()) throw std::range_error("SimpleContainerIterator::currentItem  Error: IteratorOutOfBounds");
      return d_container->at(d_current);
  }

File: pkg/Java/AmoreJ/javasrc/es/unileon/amorej/SimpleContainerIterator.java
package es.unileon.amorej;

public class SimpleContainerIterator<E extends Visible & Checkable> implements
		Iterator<E> {
	Container<E> container;
	int current;

	SimpleContainerIterator(Container<E> container) {
		this.container = container;
		this.current = 0;
	}

	@Override
	public void first() {
		this.current = 0;
	}

	@Override
	public void next() {
		this.current++;
	}

	@Override
	public boolean isDone() {
		return (this.current == this.container.size());
	}

	@Override
	public E currentItem() throws AmoreJException {
		if (this.isDone())
			throw new AmoreJException(
					"SimpleContainerIterator.currentItem  Error: IteratorOutOfBounds");
		return this.container.get(current);

	}

}
